src/helics/application_api/Federate.hpp

```diff
@@ -468,13 +458,32 @@ class Federate
     /** get the data currently stored for a particular interface handle
     @param handle the handle to get the information for
     @return a string with the data for the information*/
-    std::string const &getInfo (interface_handle handle);
+    const std::string &getInfo (interface_handle handle);
 
     /** set an interface option */
     void setInterfaceOption (interface_handle handle, int32_t option, bool option_value = true);
     /** get the current value for an interface option*/
     bool getInterfaceOption (interface_handle handle, int32_t option);
 
+    /** get the injection type for an interface,  this is the type for data coming into an interface
+    @details for filters this is the input type, for publications this is type used to transmit data, for endpoints
+    this is the specified type and for inputs this is the type of the transmitting publication
+    @param handle the interface handle to get the injection type for
+    @return a const ref to  std::string  */
+    const std::string &getInjectionType (interface_handle handle) const;
+
+    /** get the type for which data comes out of an interface,  this is the type for data coming into an interface
```

Comment:
Should this sentence not be:

get the extraction type for an interface,  this is the type for data coming out of an interface